gmqtt/aioclient.py

```diff
@@ -0,0 +1,91 @@
+import asyncio
+from .client import Client, Message
+from typing import Optional, Tuple
+
+
+class MqttClientProtocol:
+    """
+    Wraps the normal client in the async/await API
+    """
+
+    def __init__(
+        self, inner_client: Client, loop: Optional[asyncio.AbstractEventLoop] = None
+    ):
+        if loop is None:
+            loop = asyncio.get_event_loop()
+        self.loop = loop
+
+        self.cbclient = inner_client
+        self.message_queue = asyncio.Queue()
+
+    async def publish(self, topic: str, message: Message, qos=0):
+        """Publish a message to the MQTT broker"""
+        self.cbclient.publish(topic, message, qos)
+
+    async def subscribe(self, topic: str, qos) -> Tuple[str, bytes]:
+        """Subscribe to a MQTT topic and wait for a single message."""
+
+        def _on_message(client, topic, payload, qos, properties):
```

Comment:
"properties" must be passed to `queue` because they may contain useful information about message (for example - content type);

Reply:
The `queue` now stores the entire Message type now.